ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,60 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+AD.@primitive function jacobian(b::AD.EnzymeForwardBackend, f, x)
+    val = f(x)
+    if val isa Real
```

Comment:
You should also replace these with autodiff calls as well

Reply:
why?

Reply:
Forward mode gradients require repeated (potentially chunked) batching calls, as well as a re-usable shadow. This is equivalent to the inner call of jacobian. Moreover jacobian does specific post processing to create the dense matrix, which here we should avoid doing that intermediate post-processing if it needs to be separately postprocessed here anyways.

Reply:
Well this is a Jacobian function so calling the `Enzyme.jacobian` function makes sense. Perhaps we should just define the pushforward (pullback) primitive for the Enzyme Forward (Reverse) backends and let the Jacobian be constructed by AD. I just need to figure out how to do a pushforward/pullback in Enzyme. I can read the Enzyme.jacobian src.

Reply:
I defined the pullback and the Jacobian is derived from that